streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -180,8 +183,16 @@ public String apply(K key, V value) {
         Objects.requireNonNull(mapper, "mapper can't be null");
         String name = builder.newName(MAPVALUES_NAME);
 
-        builder.internalTopologyBuilder.addProcessor(name, new KStreamMapValues<>(mapper), this.name);
+        builder.internalTopologyBuilder.addProcessor(name, new KStreamMapValues<>(withKey(mapper)), this.name);
```

Comment:
we need to remove this line, too. It is taken car of in `mapValues(ValueMapperWithKey...)`

Reply:
very nice catch. Thanks.